File: Lab_Test_MNS2025031/myProg.c
// myProg.c
int modexp(int g, int a, int p) {
    long long result = 1;
    long long base = g % p;

    while (a > 0) {
        if (a & 1) result = (result * base) % p;
        base = (base * base) % p;
        a >>= 1;
    }
    return (int)result;
}

File: Lab_Test_MNS2025030/my-app/src/wasm/myProg.js
export default async function Module() {
  const response = await fetch("/wasm/myProg.wasm");
  const wasmBinary = await response.arrayBuffer();

  const wasm = await WebAssembly.instantiate(wasmBinary, {
    env: {
      abort: () => console.error("WASM aborted"),
    },
  });

  const exports = wasm.instance.exports;

  return {
    modexp: exports.c  // WASM export "c" is your modexp function
  };
}
